File: leetcode/Arrays_Hash/Contains Duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        m = {}

        for elem in nums:
            if elem not in m:
                m[elem] = 1
            else:
                m[elem] += 1
                if (m[elem] >= 2):
                    return True
        return False

File: leetcode/Arrays_Hash/Contains Duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int, int> map;
        for (int num : nums){
            if (map[num] >= 1) return true;
            else map[num]++;
        }
        return false;
    }
};